monetio/sat/nesdis_viirs_aod_aws_gridded.py

```diff
@@ -0,0 +1,319 @@
+                + satellite
+                + "/VIIRS/"
+                + satellite
+                + "_VIIRS_Aerosol_Optical_Depth_Gridded_Reprocessed/0.25_Degrees_Weekly/"
+                + year
+                + "/"
+            )
+            # Get list of all files in given year on NODD
+            all_files = fs.ls(prod_path)
+            # Loop through files, check if file date falls within observation date range
+            for file in all_files:
+                file_start = file.split("/")[-1].split("_")[7].split(".")[0].split("-")[0]
+                file_end = file.split("/")[-1].split("_")[7].split(".")[0].split("-")[1]
+                # If file within observation range, add path to list and add file size to total
+                if file_date >= file_start and file_date <= file_end:
+                    if file not in nodd_file_list:
+                        nodd_file_list.append(file)
+                        nodd_total_size = nodd_total_size + fs.size(file)
+
+    return nodd_file_list, nodd_total_size
+
+
+def open_dataset(
+    date, satellite, data_resolution="0.1", averaging_time="daily", download=False, save_path="./"
+):
+    """
+    Opens a dataset for the given date, satellite, data resolution, and averaging time.
+
+    Parameters:
+        date (str or datetime.datetime): The date for which to open the dataset.
+        satellite (str): The satellite to retrieve data from. Valid values are 'SNPP', 'NOAA20', or 'both'.
+        data_resolution (str, optional): The data resolution. Valid values are '0.050', '0.100', or '0.250'. Defaults to '0.1'.
+        averaging_time (str, optional): The averaging time. Valid values are 'daily', 'weekly', or 'monthly'. Defaults to 'daily'.
+
+    Returns:
+        xarray.Dataset: The opened dataset.
+
+    Raises:
+        ValueError: If the input values are invalid.
+    """
+    import pandas as pd
+    import s3fs
+    import xarray as xr
+
+    if satellite not in ("SNPP", "NOAA20", "both"):
+        print(
+            "Invalid input for 'satellite': Valid values are 'SNPP', 'NOAA20', 'both'. Setting default to SNPP"
+        )
+        satellite = "SNPP"
+
+    if data_resolution not in ("0.050", "0.100", "0.250"):
+        print(
+            "Invalid input data_resolution. Valid values are '0.050', '0.100', '0.250'. Setting default to 0.1"
+        )
+        data_resolution = "0.100"
+
+    if isinstance(date, str):
+        date_generated = [pd.Timestamp(date)]
+    else:
+        date_generated = [date]
+
+    # Access AWS using anonymous credentials
+    fs = s3fs.S3FileSystem(anon=True)
+
+    if averaging_time == "monthly":
+        file_list, _ = create_monthly_aod_list(satellite, date_generated, fs)
+    elif averaging_time == "weekly":
+        file_list, _ = create_weekly_aod_list(satellite, date_generated, fs)
+    else:
+        file_list, _ = create_daily_aod_list(data_resolution, satellite, date_generated, fs)
+
+    aws_file = fs.open(file_list[0])

```

Comment:
Suggest warning in the case that more than one file is available ("both" option with both satellites available)

Reply:
Maybe also an exception if `file_list` is empty, to be more informative than `file_list[0]` just failing